File: backend/components/metiers/administration_publique/utils/rbac/samples/sample_usage.test.js
// sample_usage.test.js
// Tests unitaires avancés pour les exemples RBAC JS
const rbac = require('../core/rbac_core');
const data = require('./sample_rbac_data.json');

describe('Sample usage RBAC JS', () => {
  it('vérifie la permission read', () => {
    expect(rbac.checkPermission(data.user, 'read')).toBe(true);
  });
  it('vérifie la permission inconnue', () => {
    expect(rbac.checkPermission(data.user, 'unknown')).toBe(false);
  });
  it('récupère les rôles', () => {
    expect(rbac.getUserRoles(data.user)).toContain('admin');
  });
});

File: backend/components/metiers/administration_publique/utils/rbac/samples/sample_usage.test.py
# sample_usage.test.py
"""Tests unitaires avancés pour les exemples RBAC Python"""
from ..core import rbac_core
import json
import pytest

def test_check_permission():
    with open('sample_rbac_data.json') as f:
        data = json.load(f)
    assert rbac_core.check_permission(data['user'], 'read')
    assert not rbac_core.check_permission(data['user'], 'unknown')

def test_get_user_roles():
    with open('sample_rbac_data.json') as f:
        data = json.load(f)
    assert 'admin' in rbac_core.get_user_roles(data['user'])
